### ISSUE TITLE
Very confusing code in encrypt

### ISSUE THREAD

**Issue by samuel-crypto** [pvial00/Cube#5]
I have found the following code in the encrypt function:

```
for (unsigned char byte: data) {
            sub = byte;
	    for (z=0; z < state.size(); z++) {
	        for (y=0; y < state[z].size(); y++) {
                    sub_pos = sub;
		    sub = state[z][y].at(sub_pos);
		    shift = state[z][y].at(0);
		    state[z][y].erase(state[z][y].begin()+0);
		    state[z][y].push_back(shift);
	        }
	    }
	    sub_key = key_scheduler(sub_key);
	    morph_cube(ctr, sub_key);
	    data[ctr] = char(sub);
            ctr = (ctr + 1) % alphabet_size;
}
```

What confuses me are the following:
1. sub_pos and pos are redeclared every iteration of the loop but are not used in the loop, wouldn't it be wiser to simply include them after the loop? since effectively they will contain `sub = state[state.size() - 1][state[state.size() - 1].size() - 1]`?

2. the `sub` character appears to be unaffected by the `morph_cube` function, this is intentional correct?

It would be nice to have some extra comments in the code also :)

**Comment by pvial00** [comment]
1.  I believe declaring sub_pos = sub was awkwardly placed there to solve an issue.  I had planned to revisit this.  Now is a good of time as any.  Ideally, there should be no sub_pos and only sub.

2.  Yes, the sub character is the "substituted" character and is what is returned after encryption/decryption.

**Comment by samuel-crypto** [comment]
Wonderful, let me know how it goes!

**Comment by pvial00** [comment]
sub_pos is benign and will be removed in the upcoming release.

**Comment by pvial00** [comment]
Fixed in v1.1.0.
